tunnel/winipcfg/types.go

```diff
@@ -609,31 +602,34 @@ func (obj *IPAdapterDNSSuffix) String() string {
 // AdapterName method returns the name of the adapter with which these addresses are associated.
 // Unlike an adapter's friendly name, the adapter name returned by AdapterName is permanent and cannot be modified by the user.
 func (addr *IPAdapterAddresses) AdapterName() string {
-	if addr.adapterName == nil {
-		return ""
-	}
-	slice := (*(*[maxIndexCount8]uint8)(unsafe.Pointer(addr.adapterName)))[:]
-	null := bytes.IndexByte(slice, 0)
-	if null != -1 {
-		slice = slice[:null]
+	l := 0
+	if p := unsafe.Pointer(addr.adapterName); p != nil {
+		for *((*byte)(p)) != 0 {
+			l++
+			p = addp(p, 1)
+		}
```

Comment:
This looks pretty similar to UTF16PtrToString. And there's already BytePtrFromString in sys. Can we instead just add BytePtrToString to sys, in order to complete that API family? Then we don't have to open code error prone code like this.

Reply:
Just submitted that change, if you want to give it a whirl -- https://go-review.googlesource.com/c/sys/+/263757